common/src/main/java/org/conscrypt/Hpke.java

```diff
@@ -0,0 +1,236 @@
+   * they must be decrypted in the same order as well. The API setup[Mode]Recipient must be called
+   * just once before multiple calls to this API.
+   * <p><p>
+   * Single-shot decryption:<p>
+   * If decryption a single message, the API setup[Mode]Recipient must be called
+   * before every single message. Note: Calling setup[Mode][Sender|Recipient] resets the HPKE
+   * context.
+   *
+   * @param ciphertext contains the encrypted plaintext
+   * @param aad        optional associated data
+   * @return plaintext in a byte array.
+   * @throws IllegalStateException if the API setup[Mode]Recipient hasn't been called or if an
+   *                               issue happened while performing decryption operation (an issue
+   *                               could occur most likely if the keys configured are not valid).
+   */
+  public byte[] open(byte[] ciphertext, byte[] aad) {
+    if (mSetupMode != Mode.RECIPIENT) {
+      throw new IllegalStateException("Setup recipient needs to be called before decryption");
+    }
+
+    Preconditions.checkNotNull(ciphertext, "ciphertext");
+    try {
+      return NativeCrypto.EVP_HPKE_CTX_open(mCtx, ciphertext, aad);
+    } catch (Exception e) {
+      throw new IllegalStateException(
+          "Error while decrypting with the keys provided during setup recipient", e);
+    }
+  }
+
+  /**
+   * Hybrid Public Key Encryption (HPKE) encryption.
+   * <p>
+   * Returns a {@link HpkeResult} wrapper holding the resulting encapsulated key (enc) and
+   * ciphertext given a required plaintext and an optional associated data (aad). The API has a
+   * pre-requirement, an API call to set up[Mode]Sender is needed so the API could be properly
+   * initialized. Note: This API is stateful.
+   * <p><p>
+   * Multiple messages encryption:<p>
+   * If encrypting multiple messages that are expected to be decrypted in the same sequence as how
+   * they are encrypted, the API setup[Mode]Sender must be called just once before multiple calls
+   * to this API.
+   * <p><p>
+   * Single-shot encryption:<p>
+   * If encrypting a single message, the API setup[Mode]Sender must be called
+   * before every single message. Note: Calling setup[Mode][Sender|Recipient] resets the HPKE
+   * context.
+   *
+   * @param plaintext message that will be encrypted
+   * @param aad       optional associated data
+   * @return HpkeResult wrapping the resulting encapsulated key (enc) and ciphertext.
+   * @throws IllegalStateException if the API setup[Mode]Sender hasn't been called.
+   */
+  public HpkeResult seal(byte[] plaintext, byte[] aad) {
+    if (mSetupMode != Mode.SENDER) {
+      throw new IllegalStateException("Setup sender needs to be called before encryption");
+    }
+
+    Preconditions.checkNotNull(plaintext, "plaintext");
+    final byte[] ciphertext = NativeCrypto.EVP_HPKE_CTX_seal(mCtx, plaintext, aad);
+    return new HpkeResult(mEnc, ciphertext);
```

Comment:
Why is `seal` returning `enc`? That doesn't make sense. Encryption returns just the ciphertext.
https://www.rfc-editor.org/rfc/rfc9180.html#name-encryption-and-decryption

Reply:
I have updated the code so `setupBaseSender` returns the `enc` instead and then `export` and `seal` can return just the given output. WDTY?